Mathlib/Algebra/GroupWithZero/ProdHom.lean

```diff
@@ -0,0 +1,120 @@
+/-
+Copyright (c) 2025 Yakov Pechersky. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yakov Pechersky
+-/
+import Mathlib.Algebra.Group.Prod
+import Mathlib.Algebra.GroupWithZero.WithZero
+
+/-!
+# Homomorphisms for products of groups with zero
+
+This file defines homomorphisms for products of groups with zero,
+which is identified with the `WithZero` of the product of the units of the groups.
+
+The product of groups with zero `WithZero (αˣ × βˣ)` is a
+group with zero itself with natural inclusions.
+
+TODO: Create the "GrpWithZero" category.
+
+-/
+
+namespace MonoidWithZeroHom
+
+variable (M₀ N₀ : Type*) [GroupWithZero M₀] [GroupWithZero N₀]
+
+/-- Given groups with zero `M₀`, `N₀`, the natural inclusion ordered homomorphism from
+`M₀` to `WithZero (M₀ˣ × N₀ˣ)`, which is the group with zero that can be identified
+as their product. -/
+noncomputable def inl [DecidablePred fun x : M₀ ↦ x = 0] : M₀ →*₀ WithZero (M₀ˣ × N₀ˣ) :=
```

Comment:
Out of curiosity, what's non-computable about this definition?

Reply:
Upstream noncomputable on WithZero lift and map. I have a commit removing those soon.